### PR TITLE
WIP Don't use deep pagination to fetch all entries #7139

### PR DESCRIPTION
**Pull request by rymsha**
- optimizations made reduce GC (for instance NodeBranchEntries
- CPU usage optimizations (UUID)
- Remove unused code
- added gt and lt ability for Range Filter (it was always gte and lte before)
- deep paging replaced with either scroll (to get all ids in one go) 
- or by simulating "search after" https://www.elastic.co/guide/en/elasticsearch/reference/7.x/search-request-body.html#request-body-search-search-after

there is new configuration parameter for vacuum `versionsBatchSize` which defaults to 10000 (which is equal to default index.max_result_window value in elasticsearch). without it users would not be able to reduce `index.max_result_window` value without breaking vacuum.

Vacuum should now work with any number of Versions.
Dump works with 500K Nodes (requires 1 GiB heap)
Reindex works with 500K Nodes (requires 2 GiB heap)

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

**Comment by sigdestad** [comment]
Nice work Sergey!

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/node/NodeBranchEntries.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/node/NodeBranchEntries.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/node/NodeBranchEntries.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/node/NodeBranchEntries.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

File: modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java [hunk 1]
```diff
@@ -197,25 +188,17 @@ private void dumpCommits()
...
-                nodeCommitEntries.stream().
-                    map( nodeCommitEntry -> CommitDumpEntry.create().
-                        nodeCommitId( nodeCommitEntry.getNodeCommitId() ).
-                        message( nodeCommitEntry.getMessage() ).
-                        committer( nodeCommitEntry.getCommitter() ).
-                        timestamp( nodeCommitEntry.getTimestamp() ).
-                        build() ).
-                    forEach( this.writer::writeCommitEntry );
-            }
+            final NodeCommitEntries nodeCommitEntries = this.nodeService.findCommits( nodeCommitQuery ).
```
Comment (vbradnitski) [inline modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java]: looks like   `size( NodeSearchService.GET_ALL_SIZE_FLAG )`  should be used.

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/node/NodeBranchEntries.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5176999&bid=16992317)

**Pull request merged**